ethereum_history_api/circuits/lib/src/rlp/types.nr

```diff
@@ -28,24 +28,25 @@ impl RlpFragment {
     fn assert_eq_bytes<FIELD_NAME_LEN, MAX_RLP_LEN, MAX_VALUE_LEN>(
         self,
         field_name: str<FIELD_NAME_LEN>,
-        rlp: [u8; MAX_RLP_LEN],
-        value: [u8; MAX_VALUE_LEN],
-        length: u64
+        rlp: Fragment<MAX_RLP_LEN, u8>,
+        value: Fragment<MAX_VALUE_LEN, u8>
     ) {
         assert(self.data_type == STRING, f"{field_name}: Invalid RLP type");
-        assert(self.length == length, f"{field_name}: Invalid RLP length");
-        assert(
-            sub_array_equals_up_to_length(value, rlp, self.offset, self.length), f"{field_name}: Invalid RLP value"
-        );
+        assert(self.length == value.length, f"{field_name}: Invalid RLP length");
+        assert(rlp.subfragment(self.offset, self.length).eq(value), f"{field_name}: Invalid RLP value");
     }
 
     fn assert_eq_bounded_vec<FIELD_NAME_LEN, MAX_RLP_LEN, MAX_VALUE_LEN>(
         self,
         field_name: str<FIELD_NAME_LEN>,
-        rlp: [u8; MAX_RLP_LEN],
+        rlp: Fragment<MAX_RLP_LEN, u8>,
         value: BoundedVec<u8, MAX_VALUE_LEN>
     ) {
-        self.assert_eq_bytes(field_name, rlp, value.storage, value.len);
+        self.assert_eq_bytes(
+            field_name,
+            rlp,
+            Fragment::new_with_length(value.len, value.storage)
```

Comment:
We don't have a conversion from BoundedVec? We should

Reply:
My mistake, we have it